src/InteractionModRandom.H

```diff
@@ -0,0 +1,272 @@
+                                      const PTDType& a_ptd_i, /*!< Particle tile data for infectious agent */
+                                      const PTDType& a_ptd_j, /*!< Particle tile data for susceptible agent*/
+                                      const DiseaseParm* const a_lparm,  /*!< disease paramters */
+                                      const Real a_social_scale, /*!< Social scale */
+                                      ParticleReal* const a_prob_ptr /*!< infection probability */)
+{
+    Real infect = a_lparm->infect;
+    infect *= a_lparm->vac_eff;
+
+    auto age_group_ptr_j = a_ptd_j.m_idata[IntIdx::age_group];
+    auto nborhood_ptr_i = a_ptd_i.m_idata[IntIdx::nborhood];
+    auto nborhood_ptr_j = a_ptd_j.m_idata[IntIdx::nborhood];
+    auto school_ptr_i = a_ptd_i.m_idata[IntIdx::school];
+    auto withdrawn_ptr_i = a_ptd_i.m_idata[IntIdx::withdrawn];
+    auto withdrawn_ptr_j = a_ptd_j.m_idata[IntIdx::withdrawn];
+    if (withdrawn_ptr_i[a_i] || withdrawn_ptr_j[a_j]) { return; }
+    auto random_travel_ptr_i = a_ptd_i.m_idata[IntIdx::random_travel];
+    auto random_travel_ptr_j = a_ptd_j.m_idata[IntIdx::random_travel];
+    if (random_travel_ptr_i[a_i] || random_travel_ptr_j[a_j]) {return;}
+
+    //infect *= i_mask;
+    //infect *= j_mask;
+    ParticleReal prob = 1.0_prt;
+
+    // school < 0 means a child normally attends school, but not today
+    /* Should always be in the same community = same cell */
+    if (school_ptr_i[a_i] < 0) {  // not attending school, use _SC contacts
+        prob *= 1.0_prt - infect * a_lparm->xmit_comm_SC[age_group_ptr_j[a_j]] * a_social_scale;
+    } else {
+        prob *= 1.0_prt - infect * a_lparm->xmit_comm[age_group_ptr_j[a_j]] * a_social_scale;
+    }
+    // /* Neighborhood? */
+    if (nborhood_ptr_i[a_i] == nborhood_ptr_j[a_j]) {
+        if (school_ptr_i[a_i] < 0)  {
+            // not attending school, use _SC contacts
+            prob *= 1.0_prt - infect * a_lparm->xmit_hood_SC[age_group_ptr_j[a_j]] * a_social_scale;
+        } else {
+            prob *= 1.0_prt - infect * a_lparm->xmit_hood[age_group_ptr_j[a_j]] * a_social_scale;
+        }
+    }
+
+    Gpu::Atomic::Multiply(&a_prob_ptr[a_j], prob);
+}
+
+/*! \brief Class describing agent interactions for random travel */
+template <typename AC, typename ACT, typename ACTD, typename A>
+class InteractionModRandom : public InteractionModel<AC,ACT,ACTD,A>
+{
+    public:
+
+        /*! \brief null constructor */
+        InteractionModRandom() { }
+
+        /*! \brief default destructor */
+        virtual ~InteractionModRandom() = default;
+
+        /*! \brief Simulate agent interaction for random travel */
+        void interactAgents( AC&, MultiFab&) override {}
+
+        /*! \brief Simulate agent interaction for random travel */
+        void interactAgents( AC&, MultiFab&, AC&) override ;
+
+    protected:
+
+    private:
+};
+
+/*! Simulate the interactions between agents at home and compute
```

Comment:
Can you correct the comment here? We can skip the detailed description if you want.